tests/integration/test_credentials.py

```diff
@@ -0,0 +1,106 @@
+# Copyright 2015 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"). You
+# may not use this file except in compliance with the License. A copy of
+# the License is located at
+#
+# http://aws.amazon.com/apache2.0/
+#
+# or in the "license" file accompanying this file. This file is
+# distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF
+# ANY KIND, either express or implied. See the License for the specific
+# language governing permissions and limitations under the License.
+
+import os
+import mock
+
+from botocore.session import Session
+from tests import unittest
+
+
+class TestCredentialPrecedence(unittest.TestCase):
+    def setUp(self):
+        # Clean up any existing environment
+        for name in ['AWS_ACCESS_KEY_ID',
+                     'AWS_SECRET_ACCESS_KEY',
+                     'BOTO_DEFAULT_PROFILE']:
+            if name in os.environ:
+                del os.environ[name]
+
+        # Set the config file to something that doesn't exist, then
+        # set the shared credential file to our test config.
+        os.environ['AWS_CONFIG_FILE'] = '~/.aws/config-missing'
+        Session.SessionVariables['credentials_file'] = (
```

Comment:
Same thing here.  You are modifying a class variable, which persists across tests.  Avoid doing this, as this is modifying global state and not putting it back.